src/device.rs

```diff
@@ -110,6 +110,32 @@ impl<'a> Device<'a> {
         }
     }
 
+    /// Returns the first parent with a given subsystem and devtype
+    pub fn parent_with_subsystem_devtype<T: AsRef<OsStr>, U: AsRef<OsStr>>(&self, subsystem: T, devtype: Option<U>) -> ::Result<Option<Device>> {
```

Comment:
Some comments:

* Simplifying the return type seems right to me.
* Is there any fundamental reason why Rust needs to infer a type for ```U``` if the second argument is ```None```? It seems like a type-checker would be able to handle that by eliding the ```U```, since it doesn't occur in any other part of the signature.